### PR TITLE
[HOPSWORKS-2009] Append Features to Feature Group schema

### PR DESCRIPTION
**Pull request by moritzmeister**
usage:

```scala
var houses_sold = fs.getFeatureGroup("houses_sold", 1)
val l = Seq(new Feature("asdgadfg", "double", "5.0"))
houses_sold.appendFeatures(l)
```
or
```scala
houses_sold.appendFeatures(new Feature("asdgadfg", "double", "5.0"))
```

### CODE REVIEW

**Review by SirOibaf — APPROVED** [review]
I think it's ok for the moment. If in the future we see that we are adding too many of them, we can refactor and deprecate these methods.

File: java/src/main/java/com/logicalclocks/hsfs/FeatureGroup.java [hunk 1]
```diff
@@ -211,6 +217,43 @@ public void updateStatisticsConfig() throws FeatureStoreException, IOException {
...
+   * Append a single feature to the schema of the feature group.
+   * It is only possible to append features to a feature group. Removing features is considered a breaking change.
+   *
+   * @param features
+   * @throws FeatureStoreException
+   * @throws IOException
+   */
+  public void appendFeatures(Feature features) throws FeatureStoreException, IOException {
+    List<Feature> featureList = new ArrayList<>();
+    featureList.add(features);
```
Comment (SirOibaf) [inline java/src/main/java/com/logicalclocks/hsfs/FeatureGroup.java:253]: You can use `Arrays.asList()` here.

File: java/src/main/java/com/logicalclocks/hsfs/FeatureGroup.java [hunk 2]
```diff
@@ -211,6 +217,43 @@ public void updateStatisticsConfig() throws FeatureStoreException, IOException {
...
+  /**
+   * Append features to the schema of the feature group.
+   * It is only possible to append features to a feature group. Removing features is considered a breaking change.
+   *
+   * @param features
+   * @throws FeatureStoreException
+   * @throws IOException
+   */
+  public void appendFeatures(List<Feature> features) throws FeatureStoreException, IOException {
+    featureGroupEngine.appendFeatures(this, new ArrayList<>(features));
```
Comment (SirOibaf) [inline java/src/main/java/com/logicalclocks/hsfs/FeatureGroup.java:240]: Why the `ArrayList<>` here? You already have a list.

Comment (moritzmeister) [reply java/src/main/java/com/logicalclocks/hsfs/FeatureGroup.java:240]: that list is immutable for some reason or didn't have the `addAll` method

**Pull request merged**